redbot/core/settings_caches.py

```diff
@@ -56,6 +56,54 @@ async def set_prefixes(
             await self._config.guild_from_id(gid).prefix.set(prefixes)
 
 
+class I18nManager:
+    def __init__(self, config: Config):
+        self._config: Config = config
+        self._guild_locale: Dict[int, str] = {}
+        self._guild_regional_format: Dict[int, str] = {}
+
+    async def get_locale(self, guild: Union[discord.Guild, None]) -> str:
```

Comment:
We should really be caching everything. Currently, in most cases we still end up calling config:
- in non-guild contexts we always get global locale from config
- in guild contexts, we get guild and global locale from config in cases where guild locale is `None` (which will most likely be the case for most guilds we check it for)

Reply:
Should be done.